src/nvim/buffer.c

```diff
@@ -321,8 +321,9 @@ close_buffer (
       unload_buf = true;
   }
 
-  if (buf->terminal && (unload_buf || del_buf || wipe_buf)) {
-    // terminal buffers can only be wiped
+  if ((buf->terminal || buf->b_messages)
+      && (unload_buf || del_buf || wipe_buf)) {
+    // terminal and message pane buffers can only be wiped
```

Comment:
TBH, late night brute force bug squashing.  I'll have to recheck this since I rewrote how the buffer was created after this was added.